lib/srv/db/common/auth.go

```diff
@@ -440,15 +477,94 @@ func (a *dbAuth) appendClientCert(ctx context.Context, sessionCtx *Session, tlsC
 	return tlsConfig, nil
 }
 
-// appendCAToRoot appends CA certificate from session context to provided tlsConfig.
-func appendCAToRoot(tlsConfig *tls.Config, sessionCtx *Session) (*tls.Config, error) {
+// setupTLSConfigRootCAs initializes the root CA cert pool for the provided
+// tlsConfig based on session context.
+func setupTLSConfigRootCAs(tlsConfig *tls.Config, sessionCtx *Session) error {
+	// Start with an empty pool.
+	tlsConfig.RootCAs = x509.NewCertPool()
+
+	// If CA is provided by the database object, always use it.
 	if len(sessionCtx.Database.GetCA()) != 0 {
 		if !tlsConfig.RootCAs.AppendCertsFromPEM([]byte(sessionCtx.Database.GetCA())) {
-			return nil, trace.BadParameter("invalid server CA certificate")
+			return trace.BadParameter("invalid server CA certificate")
 		}
+		return nil
 	}
 
-	return tlsConfig, nil
+	// Overwrite with the system cert pool, if required.
+	if shouldUseSystemCertPool(sessionCtx) {
+		systemCertPool, err := x509.SystemCertPool()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		tlsConfig.RootCAs = systemCertPool
+		return nil
+	}
+
+	// Use the empty pool. Client cert will be added later.
+	return nil
+}
+
+// shouldUseSystemCertPool returns true for database servers presenting
+// certificates signed by publicly trusted CAs so a system cert pool can be
+// used.
+func shouldUseSystemCertPool(sessionCtx *Session) bool {
+	// Azure Cache for Redis certificates are signed by DigiCert Global Root G2.
+	if sessionCtx.Database.IsAzure() && sessionCtx.Database.GetProtocol() == defaults.ProtocolRedis {
+		return true
+	}
+	return false
```

Comment:
```suggestion
	// Azure Cache for Redis certificates are signed by DigiCert Global Root G2.
	return sessionCtx.Database.IsAzure() && sessionCtx.Database.GetProtocol() == defaults.ProtocolRedis
```

could simplify here if you want to